etna/search/forms.py

```diff
@@ -226,20 +228,25 @@ class BaseCollectionSearchForm(forms.Form):
     )
 
     def clean(self):
-        """Collect selected filters to pass to the client in view."""
+        """
+        Overrides Form.clean() to perform additional validation on date ranges within the form.
+        """
         cleaned_data = super().clean()
 
-        try:
-            if cleaned_data.get("opening_start_date") > cleaned_data.get(
-                "opening_end_date"
-            ):
-                self.add_error(
-                    "opening_start_date",
-                    CUSTOM_ERROR_MESSAGES.get("invalid_date_range"),
-                )
-        except TypeError:
-            # Either one or both date fields are empty. No further validation necessary.
-            pass
+        opening_start = cleaned_data.get("opening_start_date")
+        opening_end = cleaned_data.get("opening_end_date")
+
+        if opening_start and opening_end and opening_start > opening_end:
+            # if both dates have valid values but invalid when together
+            self.add_error(
+                "opening_start_date",
+                ValidationError(
+                    "This date must be earlier than or equal to the 'to' date.",
+                    code="range_invalid",
```

Comment:
```suggestion
                    code="date_range_invalid",
```
Could the name be specific to dates? If not why?

Reply:
Why would an error code be field specific? It should only really be used for debugging - in which case, you'll already know which field threw it.

Reply:
There could be range between numbers as well, so I thought of specific.

Reply:
Well, that's why I changed the code to `date_range_invalid`, as it's specifically about a date range